polaris-react/.storybook/GridPanel/GridPanel.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {useMemo} from 'react';
+import {useGlobals, useGlobalTypes, useParameter} from '@storybook/api';
+import {
+  AddonPanel,
+  ArgsTable,
+  ArgTypes,
+  PresetColor,
+} from '@storybook/components';
+
+export type GlobalControlsParam = {
+  presetColors?: PresetColor[];
+};
+
+type PanelProps = {
+  active: boolean;
+};
+
+const filterUncontrolledTypes = (globalTypes: ArgTypes) =>
+  Object.entries(globalTypes).reduce((acc, [key, arg]) => {
+    if (arg.control !== undefined) acc[key] = arg;
+    return acc;
+  }, {} as ArgTypes);
+
+const addPresetColors = (globalTypes: ArgTypes, presetColors: PresetColor[]) =>
+  Object.entries(globalTypes).reduce((acc, [key, arg]) => {
+    if (arg?.control?.type !== 'color' || arg?.control?.presetColors)
+      acc[key] = arg;
+    else acc[key] = {...arg, control: {...arg.control, presetColors}};
+    return acc;
+  }, {} as ArgTypes);
```

Comment:
```suggestion
const addPresetColors = (globalTypes: ArgTypes, presetColors: PresetColor[]) =>
  Object.fromEntries(
    Object.entries(globalTypes).map(([key, arg]) => {
       if (arg?.control?.type !== 'color' || arg?.control?.presetColors) {
         return [key, arg];
        } else {
          return [key, {...arg, control: {...arg.control, presetColors}}];
        }
    })
  ) as ArgTypes;
```

Same as above, untested but encouraging use of `fromEntries()`